src/components/shared/Entity/Edit/useInitializeCollectionDraft.ts

```diff
@@ -0,0 +1,245 @@
+import { createEntityDraft } from 'api/drafts';
+import { createDraftSpec, getDraftSpecsByCatalogName } from 'api/draftSpecs';
+import {
+    getLiveSpecsByCatalogName,
+    LiveSpecsExtQuery_ByCatalogName,
+} from 'api/liveSpecsExt';
+import {
+    useBindingsEditorStore_setCollectionData,
+    useBindingsEditorStore_setCollectionInitializationAlert,
+    useBindingsEditorStore_setSchemaInferenceDisabled,
+} from 'components/editor/Bindings/Store/hooks';
+import { BindingsEditorState } from 'components/editor/Bindings/Store/types';
+import {
+    useEditorStore_persistedDraftId,
+    useEditorStore_setId,
+    useEditorStore_setPersistedDraftId,
+} from 'components/editor/Store/hooks';
+import { useCallback } from 'react';
+import { Annotations } from 'types/jsonforms';
+
+const specType = 'collection';
+
+const createGenericDraft = async (): Promise<string | null> => {
+    const draftsResponse = await createEntityDraft('Created by the UI');
+
+    return draftsResponse.data && draftsResponse.data.length > 0
+        ? draftsResponse.data[0].id
+        : null;
+};
+
+const getCollection = async (
+    collectionName: string
+): Promise<LiveSpecsExtQuery_ByCatalogName | null> => {
+    const liveSpecResponse = await getLiveSpecsByCatalogName(
+        collectionName,
+        specType
+    );
+
+    if (liveSpecResponse.data && liveSpecResponse.data.length > 0) {
+        return liveSpecResponse.data[0];
+    } else {
+        return null;
+    }
+};
+
+function useInitializeCollectionDraft() {
```

Comment:
This hook is no longer specific to edit workflows and should be relocated. Any preference on location?

I have seen hooks reasonably scoped in the _src/components/_ directory as well as housed in a meaningful subdirectory within _src/hooks/_. Wanted to use this as an opportunity to hear how you approach determining a location for _hook_ files.

Reply:
Since this is used in two components I would put this into the `src/hooks` folder